src/app/clients/getExpandedClientDetails.ts

```diff
@@ -0,0 +1,191 @@
+import supabase, { Schema } from "@/supabase";
+import {
+    familyCountToFamilyCategory,
+    formatDatetimeAsDate,
+} from "@/app/clients/getClientsTableData";
+import { Data } from "@/components/DataViewer/DataViewer";
+
+export type RawClientDetails = Awaited<ReturnType<typeof getRawClientDetails>>;
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export const getRawClientDetails = async (parcelId: string) => {
+    const response = await supabase
+        .from("parcels")
+        .select(
+            `
+        voucher_number,
+        packing_datetime,
+
+        client:clients(
+            full_name,
+            phone_number,
+            delivery_instructions,
+            address_1,
+            address_2,
+            address_town,
+            address_county,
+            address_postcode,
+
+            family:families(
+                age,
+                gender
+            ),
+
+            dietary_requirements,
+            feminine_products,
+            baby_food,
+            pet_food,
+            other_items,
+            extra_information
+        )
+
+    `
+        )
+        .eq("primary_key", parcelId)
+        .single();
+
+    return response.data;
+};
+
+export interface ExpandedClientDetails extends Data {
+    "voucher_#": Schema["parcels"]["voucher_number"];
+    full_name: Schema["clients"]["full_name"];
+    phone_number: Schema["clients"]["phone_number"];
+    packing_date: string;
+    packing_time: string;
+    delivery_instructions: Schema["clients"]["delivery_instructions"];
+    address: string;
+    household: string;
+    "Age & Gender of Children": string;
+    dietary_requirements: string;
+    feminine_products: string;
+    baby_products: Schema["clients"]["baby_food"];
+    pet_food: string;
+    other_requirements: string;
+    extra_information: Schema["clients"]["extra_information"];
+}
+
+export const rawDataToExpandedClientDetails = (
+    rawClientDetails: RawClientDetails
+): ExpandedClientDetails => {
+    if (rawClientDetails === null) {
+        return {
+            "voucher_#": "",
+            full_name: "",
+            phone_number: "",
+            packing_date: "",
+            packing_time: "",
+            delivery_instructions: "",
+            address: "",
+            household: "",
+            "Age & Gender of Children": "",
```

Comment:
is it not possible to say age_&_gender_of_children for consistency? Is the '&' messing up with it?

Reply:
I wanted to use the style that you suggested, but using the `&` within the object key introduces a syntax error